tests/aws_utils.py

```diff
@@ -358,12 +369,14 @@ def wait_for_s3_batch_job_completion(
     account_id: str,
     s3_control_client: S3ControlClient,
 ) -> DescribeJobResultTypeDef:
+    process_timeout = datetime.now() + timedelta(minutes=3)
+
     while (
         job_result := s3_control_client.describe_job(
             AccountId=account_id,
             JobId=s3_batch_job_arn,
         )
-    )["Job"]["Status"] not in S3_BATCH_JOB_FINAL_STATES:
+    )["Job"]["Status"] not in S3_BATCH_JOB_FINAL_STATES and datetime.now() < process_timeout:
```

Comment:
Should there be an `assert datetime.now() < process_timeout` instead, so that the test fails immediately if it never gets to this state?